agg-ast/ast/src/custom_serde.rs

```diff
@@ -1312,3 +1331,184 @@ impl<'de> Deserialize<'de> for Convert {
         }
     }
 }
+
+impl<'de> Deserialize<'de> for SetField {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let expression = Expression::deserialize(deserializer)?;
+        match expression {
+            Expression::Document(mut d) => {
+                let field = match d.remove("field") {
+                    Some(Expression::Literal(LiteralValue::String(s))) => s,
+                    Some(Expression::UntaggedOperator(UntaggedOperator {
+                        op: UntaggedOperatorName::Literal,
+                        args: a,
+                    })) => {
+                        if let Some(Expression::Literal(LiteralValue::String(s))) = a.first() {
+                            s.clone()
+                        } else {
+                            return Err(serde_err::custom("field to setField must be a string"));
+                        }
+                    }
+                    _ => return Err(serde_err::custom("field to setField must be a string")),
```

Comment:
The ticket mentions we should also allow `$toString` (and presumably `$convert`) expressions for the `field` argument. Is that something we'll do in a different ticket?

Reply:
Ah yes, I did skip over this and did not settle on what we shoudl actually do. The only way schema derivation can handle this properly is if the expression in a `$toString` is just a string literal. If it is non-literal, such as a field ref, then we wont be able to statically determine what the output type should be. I suppose we could just set it to Any? but that feels bad / unexpected. So this is technically incorrect for now, but I'm not sure adding it is of much value. wdyt?

Reply:
We can add a follow up ticket for this.
Reading the description field can be "any valid expression that resolves to a string constant", which means that not only $toString, but other expressions apply to, no? 
And just to make sure that I understand correctly here.  If I have a valid pipeline like `setField( field: $dateToString(dateVal), input: xx, output: xxx)`, with DateVal being a constant, we will fail deserialization instead of passing and having the field type be any?

Reply:
Yes, that is the correct interpretation of the issue